### PR TITLE
UIU-1514: add radio button set view and edit functionality

### PR DESCRIPTION
**Pull request by MaksymDryha**
Requires https://github.com/folio-org/ui-users/pull/1295
## Purpose
To be able to edit and see values of custom fields of type "Radio button set" on entity records

### Populating custom fields with their default values
 Some custom fields can have default values (like select and radio button set). It's not possible to pass these default values to a form, since `<Form initialValues={...} />` doesn't live in `stripes-smart-components`. That's why, in order to populate fields with default values, I imperatively change their values after there are rendered.
### Supporting final-form and redux-form
To imperatively change the value of a field it's necessary to know if its a part of `final-form` or `redux-form`. For this purpose, a couple of props were added:
### redux-form props
* `isReduxForm` - used to tell custom fields if they are being rendered inside `redux-form`. If `isReduxForm` is `true`, then custom fields will dispatch `redux-form`'s `change` action to populate fields with default values
* `formName` - when dispatching the `change` action, we need to know the name of the form whose value will be changed

### final-form props
If `isReduxForm` is `false`, then the following props should be passed:
* `changeFormValue` - a function used to change `final-form` field value. We can simply pass `form.change` function passed with `Form` render props
* `finalFormCustomFieldsValues` - custom fields values stored in final-form state. Can be retrieved using `Form` render props:  {form.getState().values.customFields}. 

## Example of using with final-form
```js
<Form ...>
  {({form: { change, getState } }) => (
    <EditCustomFieldsRecord
      changeFormValue={change}
      finalFormCustomFieldsValues={getState().values.customFields}
      accordionId="customFields"
      onToggle={this.toggleSection}
      expanded
      backendModuleName="mod-kb-ebsco-java"
      entityType="package"
      fieldComponent={Field}
    />
  )}
</Form>

```

## Example of using with redux-form
```js
<EditCustomFieldsRecord
  isReduxForm
  formName="userForm"
  accordionId="customFields"
  onToggle={this.handleSectionToggle}
  expanded={sections.customFields}
  backendModuleName="users"
  entityType="user"
  fieldComponent={Field}
/>
```

## Screenshot
image

### CODE REVIEW

File: lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js [hunk 1]
```diff
@@ -57,8 +77,14 @@ const EditCustomFieldsRecord = ({
   backendModuleId,
   backendModuleName,
   entityType,
-  fieldComponent,
   columnCount,
+  dispatch,
```
Comment (BogdanDenis) [inline lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js:81]: Where does this component get access to ‘dispatch’ from?
I think it’ll be better if we’re not giving the component access to ‘dispatch’ directly but bind actions in ‘mapDispatchToProps’

File: lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js [hunk 2]
```diff
@@ -100,29 +182,41 @@ const EditCustomFieldsRecord = ({
...
       });
+    } else if (customField.type === fieldTypes.RADIO_BUTTON_SET) {
+      return createCustomFieldRenderFunction(customField)({
+        children: getRadioButtonSetChildren(customField),
+        label: (
+          <Label required>
+            {renderCustomFieldLabel(customField)}
+          </Label>
+        ),
+        required: true,
```
Comment (BogdanDenis) [inline lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js:218]: Are all radio set fields required by default?

**Comment by MaksymDryha** [comment]
> Looks great! Could you please update changelog and documentation with this description, because it could be difficult for some developers to find how working with redux form or final-form

The docs definitely have to be updated, good point, thank you!

File: lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js (hunk 1 above)
Comment (MaksymDryha) [reply lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js]: @BogdanDenis , makes sense!

File: lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js [hunk 3]
```diff
@@ -88,8 +113,65 @@ const EditCustomFieldsRecord = ({
...
+      });
+    };
+
+    if (customFieldsLoaded && sectionTitleLoaded) {
+      initializeFields();
+    }
+  }, [
+    customFieldsLoaded,
+    sectionTitleLoaded,
+    changeFinalFormField,
```
[the remark this answers is not in the document]
Comment (MaksymDryha) [reply lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js]: @BogdanDenis, I would prefer to follow the rule that encourages us to not lie about effect dependencies

File: lib/CustomFields/pages/EditRecordCustomFields/EditRecordCustomFields.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+          const fieldIsEmpty = !hasIn(finalFormCustomFieldsValues, cf.refId);
+
+          if (fieldHasOptions && fieldIsEmpty) {
+            const valueToSet = cf.selectField.defaults[0];
+
+            changeFinalFormField(`customFields.${cf.refId}`, valueToSet);
+          }
+        }
+      });
+    };
```
Comment (Vladyslav-Velytskyi) [inline lib/CustomFields/pages/EditRecordCustomFields/EditRecordCustomFields.js:142]:
```suggestion
          const fieldIsEmpty = !hasIn(reduxFormCustomFieldsValues, cf.refId);

          if (fieldHasOptions && fieldIsEmpty) {
            const valueToSet = cf.selectField.defaults[0];

            if (isReduxForm) {
              changeReduxFormField(`customFields.${cf.refId}`, valueToSet);
            } else {
              changeFinalFormField(`customFields.${cf.refId}`, valueToSet);
            }
          }
      });
    };
```

little bit hard coded, you duplicate yourself

Comment (MaksymDryha) [reply lib/CustomFields/pages/EditRecordCustomFields/EditRecordCustomFields.js]: @Vladyslav-Velytskyi , we have to know if what type of form is used before calling 
`const fieldIsEmpty = !hasIn(reduxFormCustomFieldsValues, cf.refId);
`
or
`
const fieldIsEmpty = !hasIn(finalFormCustomFieldsValues, cf.refId);
`

Comment (Vladyslav-Velytskyi) [reply lib/CustomFields/pages/EditRecordCustomFields/EditRecordCustomFields.js]: oh, yes, you're right, sorry, i missed that first arguments are different

File: lib/CustomFields/constants.js [hunk 5]
```diff
@@ -99,6 +100,7 @@ export const fieldComponents = {
   [fieldTypes.TEXTAREA]: TextArea,
   [fieldTypes.CHECKBOX]: Checkbox,
   [fieldTypes.SELECT]: Select,
+  [fieldTypes.RADIO_BUTTON_SET]: RadioButtonGroup,
```
[the remark this answers is not in the document]
Comment (MaksymDryha) [reply lib/CustomFields/constants.js]: Makes sense! I added a commit with this change

**Pull request merged**